src/main/java/com/github/manolo8/darkbot/core/utils/EntityList.java

```diff
@@ -82,11 +83,12 @@ public EntityList(Main main, EventBrokerAPI eventBroker) {
 
         this.entityRegistry.addToAll(entity -> {
             if (entity instanceof Obstacle) obstacles.add((Obstacle) entity);
-            eventBroker.sendEvent(new EntityCreateEvent(entity));
         });
-        this.entityRegistry.addDefault(e -> {
-            unknown.add(e);
-            eventBroker.sendEvent(new EntityCreateEvent(e));
+        this.entityRegistry.addDefault(unknown::add);
+
+        this.entityRegistry.onAnyEntityCreate(entity -> {
```

Comment:
is there any reason entityRegistry isn't created with all these directly? i'm not sure why we're doing these .setMain, .addToAll etc
can't we just `new EntityRegistry(main, this::onEntityCreate, unknown::add);`
and have an onEntityCreate method here that adds to all, adds to obstacle (if needed), and sends the event?